### PR TITLE
Dual wielding guns

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
If you are holding two guns you will fire both guns when you click. Shots are slightly staggered like burst fire. Only works with harm intent.

:cl: Kor
rscadd: You can now dual wield guns. Firing a gun will automatically fire the gun held in your off hand if you are on harm intent.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
DUAL BERETTAS
"SS13: GLOBAL OFFENSIVE" MEME CONTINUES TO PROPAGATE

**Comment by GunHog** [comment]
Not really sure on this one :|

**Comment by xxalpha** [comment]
It should alternate instead of firing at the same time.

**Comment by Supermichael777** [comment]
taze + lase
I LOVE IT.

**Comment by KorPhaeron** [comment]
@xxalpha 

You can already alternate by hitting X

**Comment by Supermichael777** [comment]
only if holding two stechkins makes you leap everywhere.

**Comment by Cebasiz** [comment]
mode should probably be toggleable (alternating or both at the same time)

might actually be pretty neat.

**Comment by KorPhaeron** [comment]
Not sure where you'd toggle such a thing. Dont want to add more intents and stuff just for this

@PKpenguin321

Good catch. I'll make it check

**Comment by WJohn** [comment]
YEEEESSSSSS

I've been wanting this for years!

**Comment by Dorsisdwarf** [comment]
make it so you cant dual wield Heavy guns

**Comment by Xhuis** [comment]
ATATATATATATATA

**Comment by xxalpha** [comment]
In before armor buffs.

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -158,6 +158,21 @@
 		if(!can_trigger_gun(L))
 			return
 
+	//DUAL WIELDING
+	var/obj/item/weapon/gun/off_hand
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.r_hand == src && H.l_hand && istype(H.l_hand, /obj/item/weapon/gun))
```
Comment (RemieRichards) [inline code/modules/projectiles/gun.dm]: dont check for ```l_hand```, if it isn't there ```istype()``` fails anyway.

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -158,6 +158,21 @@
...
+			off_hand = H.l_hand
+
+		else if(H.l_hand == src && H.r_hand && istype(H.r_hand, /obj/item/weapon/gun))
```
Comment (RemieRichards) [inline code/modules/projectiles/gun.dm]: dont check for ```r_hand```, if it isn't there ```istype()``` fails anyway.

**Comment by GunHog** [comment]
>In before armor buffs.

Not a bad idea after they got turned to paper in that huge nerf.

**Comment by Shadowlight213** [comment]
:-1:

**Comment by KorPhaeron** [comment]
>needs reproducing/testing

I already tested this though it works

**Comment by KorPhaeron** [comment]
>make it so you cant dual wield Heavy guns

You can't fire those unless your offhand is free though I don't need to add checks for that.

**Comment by Durkel** [comment]
As cool as this sounds, I think it might make using multiple weapons clunky otherwise you'll waste ammo on something like a laser/taser combo. Unless it only works for something like the stechkin because that's honestly the only pair the makes sense anyway. Good luck dual wielding two 357s and actually hitting something.

**Comment by KorPhaeron** [comment]
I guess I could make it only work on harm intent, but you have to avoid using harm intent on guns to avoid that awful SLOWLY POINTS AT HEAD thing when you're trying to kill someone

**Comment by Supermichael777** [comment]
grab intent without open hands?
i mean your acting like theres only 2 intents
intent system is crapy anyway.

**Comment by CollenN** [comment]
I'd make it toggleable, there are times where you wouldn't want to be firing both of them at once, such as if you're trying to conserve ammo, or if you're just trying to stun someone and you have a laser gun in your off-hand.

**Comment by Amemelius** [comment]
Make it toggleable. There are times you want to only use one, and others you want to use both. For instance, two eguns disabler or laser spamming? Both. Taser and a laser gun for quick taze and laze? Only one.

**Comment by Fox-McCloud** [comment]
>make it so you cant dual wield Heavy guns

This, so much

**Comment by RemieRichards** [comment]
Heavy weapons need the other hand to be empty to fire, meaning this should already be covered.

**Comment by Fox-McCloud** [comment]
Are you sure about that Remie? I thought they only had a *chance* to throw the gun out of your hands?

**Comment by KorPhaeron** [comment]
You'll have nausea inducing recoil from both guns and both rolling to be thrown out of your hands, it'd be terrible

**Comment by Deanthelis** [comment]
You'll have nausea inducing recoil from both guns and both rolling to be thrown out of your hands, it'd be terrible

New Genetics mutation. Ramboism, safely wield heavy weapons in one hand. Spontaneously grows a tie around your head, cannot be removed, but does not occupy any slots.

**Comment by tkdrg** [comment]
> Harm intent only

This is just confusing to me, but whatever.

**Comment by KorPhaeron** [comment]
I predict this gets reverted for being hideously unbalanced within a week or so.

That or I make it some dumb DUALWIELDER var to lock it to certain antagd

**Pull request merged**